### ISSUE TITLE
publish release script is broken

### ISSUE THREAD

**Issue by niklasad1** [paritytech/subxt#1618]
The publish release script: https://github.com/paritytech/subxt/blob/master/scripts/publish.sh is both broken and outdated.

1. It needs to be updated to support workspace dependencies
2. It needs to be updated for the new crates added such as `core`, `light-client` and `signer`
3. Because `core` has the signer as a dev-dependency it needs to be removed otherwise it can't be released.

**Comment by jsdw** [comment]
oh I forgot we even had such a script and just copy the command in the RELEASING doc. We def don't need a bunch of the ocmplexity there (ie cargo now waits for the crate itself anyway) so I'd be game for removing or just simplifying it! (perhaps it could handle using `cagro hack` or whatever if we can't otherwise resolve that cyclic dev dependency issue

**Comment by niklasad1** [comment]
Unfortunately `cargo hack` didn't work with hacks.

```bash
➜ core (master) ✗ cargo hack publish --allow-dirty --remove-dev-deps --dry-run
info: running `cargo publish --allow-dirty --dry-run` on subxt-core (1/1)
error: failed to load manifest for workspace member `/home/niklasad1/Github/subxt/testing/integration-tests`
referenced by workspace at `/home/niklasad1/Github/subxt/Cargo.toml`

Caused by:
  failed to parse manifest at `/home/niklasad1/Github/subxt/testing/integration-tests/Cargo.toml`

Caused by:
  feature `unstable-light-client` includes `subxt/unstable-light-client`, but `subxt` is not a dependency
``` 

Thus, I think `cargo hack` removes all dev-deps and not only the ones in the core crate...

**Comment by lexnv** [comment]
I believe it would be easier to copy paste the release command from the RELEASING doc. Then, we won't have to maintain anymore the release script, just the doc command. What do you guys think?

**Comment by niklasad1** [comment]
Yes, but there is the caveat with the cyclic dependencies for `subxt-core` such that we can't release it without comment out the dev-dependecies.

So I'm in favor of removing it, since the what we have in the RELEASE doc is sufficient
